wrapanapi/systems/redfish.py

```diff
@@ -35,3 +51,84 @@ def info(self):
 
     def __del__(self):
         """Disconnect from the API when the object is deleted"""
+
+    def server_stats(self, *requested_stats, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the stats
+        requested_stats = requested_stats or self._stats_available
+
+        return {stat: self._server_stats_available[stat](self, requester)
+                for stat in requested_stats}
+
+    def server_inventory(self, *requested_items, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the inventory
+        requested_items = requested_items or self._server_inventory_available
+
+        return {item: self._server_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def get_server(self, server_name):
+        """
+        Fetch a Redfish resource that represents a particular physical server.
+
+        Note: ideally, we would have used a Redfish oid of the server that the
+        Redfidh provider is currently storing and showing in the GUI labeled
+        with "UUID". However, we are not clear with how to get that information
+        from the appliance.
+        """
+        for node in self.api_client.Systems.Members:
+            # Redfish representation of a computation system doesn't have a
+            # name, and the provider uses the Id of the system as a name
+            if node.Id == str(server_name):
+                return node
+
+    def num_servers(self):
+        return len(self.api_client.Systems.Members)
+
+    def get_server_cores(self, server_name):
+        physical_system = self.get_server(server_name)
+
+        num_cores = 0
+        for proc in physical_system.Processors.Members:
+            num_cores += int(proc.TotalCores)
+
+        return num_cores
```

Comment:
Nitpick: this can be simplified into something similar to `sum([int(p.TotalCores) for p in physical_system.Processors.Members])`.

Reply:
Thanks, I applied the pattern you suggested.